app/src/main/kotlin/zeta/android/apps/ui/activity/NavigationActivity.kt

```diff
@@ -0,0 +1,190 @@
+
+    private var mViews: Views? = null
+
+    @Inject lateinit var mPresenter: NavigationPresenter
+
+    internal class Views(root: AppCompatActivity) : BaseViews(root.findViewById(R.id.zeta_drawer_layout)) {
+
+        @BindView(R.id.zeta_drawer_layout)
+        lateinit var drawerLayout: DrawerLayout
+
+        @BindView(R.id.zeta_app_bar_layout)
+        lateinit var appBarLayout: AppBarLayout
+
+        @BindView(R.id.zeta_toolbar)
+        lateinit var toolbar: Toolbar
+
+        @BindView(R.id.zeta_nav_view)
+        lateinit var navigationView: NavigationView
+
+        @BindView(R.id.container)
+        lateinit var fragmentContainer: View
+
+        val headerImageView: ImageView
+
+        val headerTitle: TextView
+
+        val headerEmail: TextView
+
+        init {
+            val headerView = navigationView.getHeaderView(0)
+            headerImageView = headerView.findViewById(R.id.header_image_view) as ImageView
+            headerTitle = headerView.findViewById(R.id.header_title) as TextView
+            headerEmail = headerView.findViewById(R.id.header_email) as TextView
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        AndroidInjection.inject(this)
+        setTheme(R.style.AppTheme)
+        configureTaskDescription()
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_navigation)
+
+        mViews = Views(this)
+        setSupportActionBar(mViews!!.toolbar)
+
+        val supportFragmentManager = supportFragmentManager
+        mNavigationFragmentManager.setFragmentManager(supportFragmentManager)
+        mNavigationFragmentManager.setContainerId(R.id.container)
+        mNavigationFragmentManager.setDrawerLayout(mViews!!.drawerLayout)
+        mNavigationFragmentManager.setDrawer(mViews!!.navigationView)
+
+        val toggle = ActionBarDrawerToggle(
+                this, mViews!!.drawerLayout, mViews!!.toolbar,
+                R.string.navigation_drawer_open,
+                R.string.navigation_drawer_close)
+        mViews!!.drawerLayout.addDrawerListener(toggle)
+        toggle.syncState()
+
+        mPresenter.onCreate(this)
+
+        mViews!!.navigationView.setNavigationItemSelectedListener(this)
+
+        if (savedInstanceState == null) {
+            // mNavigationFragmentManager.addAsBaseFragment(HomeFragment.newInstance());
+        }
+    }
+
+    override fun onPostResume() {
+        super.onPostResume()
+        mPresenter.onPostResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        mPresenter.onPause()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        ZetaApplication.watchForMemoryLeaks(applicationContext, this)
+        mPresenter.onDestroy()
+        mViews!!.clear()
+        mViews = null
+    }
+
+    override fun onBackPressed() {
+        if (mViews != null && mViews!!.drawerLayout.isDrawerOpen(GravityCompat.START)) {
```

Comment:
does it work to do `mViews?.drawerLayout.isDrawerOpen(GravityCompat.START) ?: false`
Pardon my unfamiliarity with the allowed syntax.